app/models/workshop.rb

```diff
@@ -98,21 +94,45 @@ def location
     host.address.city rescue ''
   end
 
+  def time_zone
+    chapter.time_zone
+  end
+
   def self.policy_class
     WorkshopPolicy
   end
 
+  def date_and_time
+    return nil unless super
+    super.in_time_zone(time_zone)
+  end
+
+  def rsvp_opens_at
+    return nil unless super
+    super.in_time_zone(time_zone)
+  end
+
   def date
     I18n.l(date_and_time, format: :dashboard)
   end
 
+  def time
+    date_and_time.try(:time)
+  end
+
   private
 
-  def combine_date_and_time
-    self.date_and_time = date_and_time.change(hour: time.hour, min: time.min)
+  def set_datetime_fields
+    new_date_and_time = datetime_from_fields(local_date, local_time)
+    self.date_and_time = new_date_and_time if new_date_and_time
+    new_opens_at = datetime_from_fields(rsvp_open_local_date, rsvp_open_local_time)
+    self.rsvp_opens_at = new_opens_at if new_opens_at
   end
```

Comment:
This method would benefit from splitting into several methods, maybe one for setting `rsvp_opens_at` and one for setting `date_and_time`

Reply:
Good point